### PR TITLE
Custom Slack settings with validation

### LINKED ISSUE #1541: Fancier looking Slack integration settings page
You know what I mean right ;)

### PR DESCRIPTION
**Pull request by tlrobinson**
Resolves #1541

### CODE REVIEW

File: frontend/src/admin/settings/components/SettingsEditor.jsx [hunk 1]
```diff
@@ -63,7 +64,18 @@ export default class SettingsEditor extends Component {
...
                         elements={section.settings}
                         updateEmailSettings={this.props.updateEmailSettings}
-                        sendTestEmail={this.props.sendTestEmail} />
+                        sendTestEmail={this.props.sendTestEmail}
+                    />
+                </div>
+            );
+        } else if (section.name === "Slack") {
+            return (
+                <div className="MetadataTable px2 flex-full">
```
Comment (kdoh) [inline frontend/src/admin/settings/components/SettingsEditor.jsx]: any particular reason we're using ```flex-full``` here?

**Comment by agilliland** [comment]
I'm not getting a heart in the emoji.  Not sure if that's intentional or not.

screen shot 2015-11-24 at 10 10 03 am

**Comment by agilliland** [comment]
Apart from the heart emoji everything worked well for me & the clojure code looks g2g :+1:

**Comment by kdoh** [comment]
Looking good @tlrobinson! There are just a couple things missing from the design side:

- [ ] button needs an external link icon (like the one from the emails)  
- [ ] we should set a max width on the setting body to make sure the instructive text has a readable line length
- [ ] doesn't look like the urls are updating properly when you switch between the settings pages
- [ ] there's extra somewhat redundant descriptive text below the label and above the token field
- [ ] the token field should say "Enter the token you received from Slack" instead of "a bunch of crazy letters and numbers"

Here's the invision link if you need it for reference.
https://invis.io/XF4VQ0Q75

**Comment by tlrobinson** [comment]
"doesn't look like the urls are updating properly when you switch between the settings pages"

This has never been implemented.

Do we have a path for the external icon?

I'll take a stab at the rest of it.

**Comment by kdoh** [comment]
Ah. I thought it was working via a url param as part of the work you had done for linking people to Slack setup for Pulse onboarding. If its not as easy as having the sidebar buttons link to that url + the param then no worries and we can tackle that as a separate issue.

**Comment by agilliland** [comment]
The urls are not actually setup to be route changes per section, they have always just been a way to start on a specific section.

**Comment by kdoh** [comment]
Aren't the two equivalent? Its not a big deal either way but it seems like that'd be a decent way to get the url to be "right" until we clean up the urls and move to a ```/settings/slack``` style format unless that's gonna prevent data loading or something like that.

Anyhoo, that's bottom of the list compared to the other stuff.

**Comment by tlrobinson** [comment]
Yeah, it's just what we have now was quicker to implement. The "correct" way isn't a lot harder but we should do it as a separate PR.

**Comment by tlrobinson** [comment]
Should be ready to go.

**Comment by kdoh** [comment]
:+1:

**Pull request merged**